Fix yet another emulator precision indexing bug

dmcdougall opened the pull request:
Although this could have been avoided, please merge #631 before merging this PR.

dmcdougall committed:
Add logic to compute number of nonzero evals

dmcdougall committed:
Fix logic in truncation error precision use

We want to calibrate the truncation error precision when the number of
svd basis vectors we use is less than the number of "positive" singular
values, not when it is less than the number of outputs.

dmcdougall committed:
Fix yet another emulator precision indexing bug

roystgnr commented on src/gp/src/GPMSA.C:
```diff
@@ -401,7 +401,7 @@ GPMSAEmulator<V, M>::lnValue(const V & domainVector,
 
       const double emulator_precision =
-        domainVector[dimParameter+1];
+        domainVector[dimParameter+(num_svd_terms<num_nonzero_eigenvalues)];
```
I think this code is wrong, but I don't think this is the right fix here.  This is the lambda_eta term, isn't it?  So if num_svd_terms < num_nonzero_eigenvalues then we should be indexing with dimParameter; if not then we shouldn't be adding the inv(lambda_eta * K^T * K) term at all!

dmcdougall replied:
Oh, I see.  This is the truncation error precision, not the emulator precision.  In that case I'll change the name of that variable as well as it's misleading.

roystgnr replied:
Please do, and sorry for the confusion.

dmcdougall committed:
Add logic for conditionally adding KT_K^-1 term

We only add the KT_K_inv term if we're acutally calibrating the
truncation error precision term.

roystgnr commented:
:+1: when Travis is happy
